src/skdh/io/csv.py

```diff
@@ -0,0 +1,346 @@
+"""
+CSV reading process
+
+Lukas Adamowicz
+Copyright (c) 2023. Pfizer Inc. All rights reserved
+"""
+from warnings import warn
+
+from numpy import tile, arange, mean, diff, asarray, argmin, abs, vstack, unique, all as npall, int_
+from pandas import read_csv, to_datetime, to_timedelta, Timedelta
+
+from skdh.base import BaseProcess
+from skdh.io.base import check_input_file
+
+
+def handle_timestamp_inconsistency(df, fill_gaps, accel_col_names, accel_in_g, g):
+    """
+    Handle any time gaps, or timestamps that are only down to the second.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        Dataframe containing timestamps and acceleration data.
+    fill_gaps : bool
+        Fill data gaps with the vector [0, 0, 1] or [0, 0, `g`] depending on
+        accel units.
+    accel_col_names : array-like
+        Array-like (size 3) of the acceleration column names in XYZ order.
+    accel_in_g : bool
+        If acceleration values are already in units of "g".
+    g : float
+        Gravitational acceleration in m/s^2.
+
+    Returns
+    -------
+    df : pandas.DataFrame
+        Dataframe with corrected timestamps.
+    fs : float
+        Number of samples per second.
+    """
+    # get a sampling rate. If non-unique timestamps, this will be updated
+    n_samples = mean(1 / diff(df['_datetime_'][:2500]).astype(int)) * 1e9  # datetime diff is in ns
+
+    # first check if we have non-unique timestamps
+    nonuniq_ts = df["_datetime_"].iloc[1] == df["_datetime_"].iloc[0]
```

Comment:
aren't we only comparing the first and second timestamps here? There might be timestamps that are different later in the series too, no?

Reply:
Yeah, but the point is to catch if there are blocks of timestamps that are all the same (ie ActiGraph has blocks of 1s that all have the same timestamp). If we have timestamps that are not unique after this check, it will fail to read because it wouldn't know how to handle that case.